### PR TITLE
fix empty iop read from stdin for operator.

### LINKED ISSUE #29930: proxyMetadata cannot be set with istioctl install; manifest generate works
```
$ cat <<EOF | istioctl install -y -d manifests/ -f -
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    defaultConfig:
      proxyMetadata:
        ISTIO_META_DNS_CAPTURE: "true"
        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
EOF
✔ Istio core installed
✔ Istiod installed
✔ Ingress gateways installed
✔ Installation complete                     
$ ki get cm istio -oyaml | k neat
apiVersion: v1
data:
  mesh: |-
    defaultConfig:
      discoveryAddress: istiod.istio-system.svc:15012
      proxyMetadata:
        DNS_AGENT: ""
      tracing:
        zipkin:
          address: zipkin.istio-system:9411
    enablePrometheusMerge: true
    rootNamespace: istio-system
    trustDomain: cluster.local
  meshNetworks: 'networks: {}'
```

You can see the proxy metadata is dropped

But using generate works fine:

```
$ cat <<EOF | istioctl manifest generate -d manifests/ -f - | k grep ConfigMap/istio
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    defaultConfig:
      proxyMetadata:
        ISTIO_META_DNS_CAPTURE: "true"
        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
EOF

apiVersion: v1
kind: ConfigMap
metadata:
  name: istio
  namespace: istio-system
data:
  mesh: |-
    defaultConfig:
      discoveryAddress: istiod.istio-system.svc:15012
      proxyMetadata:
        DNS_AGENT: ""
        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
        ISTIO_META_DNS_CAPTURE: "true"
      tracing:
        zipkin:
          address: zipkin.istio-system:9411
    enablePrometheusMerge: true
    rootNamespace: istio-system
    trustDomain: cluster.local
```

### PR DESCRIPTION
**Pull request by morvencao**
Please provide a description for what this PR is for.
resolve: https://github.com/istio/istio/issues/29930

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: releasenotes/notes/29930.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+  - 29930
+
+releaseNotes:
+- |
+  **Fixed** `istioctl install` read empty istiooperator from standard input.
```
Comment (ericvn) [inline releasenotes/notes/29930.yaml:9]: I'm not sure we need a release note if the fix is for something that wasn't shipped. At least I think what is fixed is correct in the 1.8 branch, and this was a regression from 1.8, and never shipped. That said, I'm fine leaving this in.

Comment (morvencao) [reply releasenotes/notes/29930.yaml]: Thanks @ericvn
removed the release note.

**Review by howardjohn — COMMENTED** [review]
Can you add a test?

**Review by howardjohn — APPROVED** [review]
approving to unblock docs, but we need a test - this is easy to break

**Comment by ericvn** [comment]
A test that was fixed by this PR in `master` in failing in `release-1.9`. This PR should be in 1.9. I'm not sure if something regressed at this point or not.

**Pull request merged**